firefox-ios/Client/Frontend/Browser/DownloadQueue.swift

```diff
@@ -283,6 +319,20 @@ class DownloadQueue {
             downloadProgress[uuid] = nil
         }
 
+        if let hasContentEncoding = download.hasContentEncoding,
+           hasContentEncoding,
+           let download = download as? HTTPDownload {
+            // We need to do a quick HEAD request to the server to check the uncompressed file size. Otherwise, the download
+            // progress will appear to exceed the `totalBytesExpected`. [FXIOS-9039]
+            // This is because NSURLSession automatically decodes gzip/compress encoded content and reports the decoded
+            // bytes to the delegate methods.
+            Task {
+                if let size = await download.getUncompressedSize(), size != download.totalBytesExpected {
+                    downloadProgress[uuid]?.totalExpected = size
+                }
+            }
```

Comment:
What is the outcome if for some reason this HEAD request stalls or times outs? It looks like the download is kicked off regardless below by `resume()`, and so users would either just see A) the incorrect file size (which is what happens already) or B) they'd see the incorrect size briefly and then it would self-correct once the totalExpected size was updated? Am I reading that accurately?